spec/models/report_spec.rb

```diff
@@ -0,0 +1,101 @@
+require 'rails_helper'
+
+RSpec.describe Report, type: :model do
+  let(:user){
+    FactoryBot.create :user
+  }
+  let(:user_1){
+    FactoryBot.create :user
+  }
+  let(:user_2){
+    FactoryBot.create :user
+  }
+  let(:department){
+    FactoryBot.create :department
+  }
+
+  let(:department_1){
+    FactoryBot.create :department
+  }
+
+  let(:department_2){
+    FactoryBot.create :department
+  }
+  let(:report){
+    FactoryBot.create :report
+  }
+
+  let(:report_1){
```

Comment:
phân biệt let và let!

Reply:
LET: chỉ có giá trị trong từng example, nếu có gọi tới nhiều lần thì cũng chỉ khai báo 1 lần duy nhất, giá trị bên trong let sau khi gọi ra đã được cache lại nên không tác động thay đổi được trong example
Let!: được gọi trong 1 before ngầm, nghĩa là được thực thi và cache lại trước khi khối được thực thi, hàm này hữu ích khi cần định nghĩa lại trước khi khối it được thực thi